src/xpath.c

```diff
@@ -434,12 +434,12 @@ cast_string_recursive(struct lyd_node *node, struct lys_module *local_mod, int f
                 LY_CHECK_ERR_RETURN(!buf, LOGMEM(local_mod->ctx), -1);
                 break;
             case LYD_ANYDATA_DATATREE:
-                if (lyd_print_mem(&buf, any->value.tree, LYD_XML, LYP_WITHSIBLINGS)) {
+                if (!lyd_print_mem(&buf, any->value.tree, LYD_XML, LYP_WITHSIBLINGS)) {
```

Comment:
This was actually correct, this printer does not return number of printed characters but 0 on success, non-zero on error.

Reply:
OK, I will update it.